docs/rollup-boost-ha.md

```diff
@@ -0,0 +1,245 @@
+  - Default setting for normal external block production.
+
+- `DryRun`  
+  - `rollup-boost` forwards all Engine API requests to both the builder and default execution client.
+  - Builder payloads are validated with the local execution client but the default execution client block will always be returned to `op-node` to propagate to the network.
+  - Useful during deployments, dry runs, or to validate builder behavior without publishing builder blocks to the network.
+
+- `Disabled`
+  - `rollup-boost` does not forward any Engine API requests to the builder.
+  - Block construction is handled exclusively by the default execution client.
+  - Useful as an emergency shutoff switch in the case of critical failures/emergencies.
+
+```rust
+pub enum ExecutionMode {
+    /// Forward Engine API requests to the builder, validate builder payloads and propagate to the network 
+    Enabled,
+    /// Forward Engine API requests to the builder, validate builder payloads but
+    /// fallback to default execution payload
+    DryRun,
+    // Do not forward Engine API requests to the builder 
+    Disabled,
+}
+```
+
+<br>
+
+## Debug API
+
+`rollup-boost` exposes a Debug API that allows operators to inspect and modify the current execution mode at runtime without restarting the service. This provides flexibility to dynamically enable, disable, or dry-run external block production based on builder behavior or network conditions. The Debug API is served over HTTP using JSON RPC and consists of the following endpoints:
+
+### `debug_setExecutionMode`
+
+Sets the current execution mode for `rollup-boost`.
+
+**Request**:
+
+```
+{
+  "method": "debug_setExecutionMode",
+  "params": [ "enabled" | "dry_run" | "disabled" ],
+  "id": 1,
+  "jsonrpc": "2.0"
+}
+```
+
+**Response**:
+
+```
+{
+  "result": null,
+  "id": 1,
+  "jsonrpc": "2.0"
+}
+```
+
+### `debug_getExecutionMode`
+
+Retrieves the current execution mode.
+
+**Request**:
+
+```
+{
+  "method": "debug_getExecutionMode",
+  "params": [],
+  "id": 1,
+  "jsonrpc": "2.0"
+}
+```
+
+**Response:**
+
+```
+{
+  "result": "enabled" | "dry_run" | "disabled",
+  "id": 1,
+  "jsonrpc": "2.0"
+}
+```
+
+<br>
+
+## Failure Scenarios
+
+Below is a high level summary of how each failure scenario is handled. All existing failure modes assumed by upstream `op-conductor` are maintained:
+
+| Failure Scenario | Category | Scenario and Solution |
+| --- | --- | --- |
+| Leader Sequencer Fails | Sequencer Failure | Conductor will detect sequencer failure and start to transfer leadership to another node, which will start sequencing instead. `op-conductor` will prioritize electing a leader with full health however if there are no sequencers that are fully healthy, a sequencer with partial health will be selected. Once the originally failed sequencer is back online, it will join the sequencer set as a follower. |
```

Comment:
same as above, suggest not to ask conductor to prioritize certain sequencers for the reason mentioned